server/app/controllers/SessionController.java

```diff
@@ -0,0 +1,38 @@
+package controllers;
+
+import auth.ProfileUtils;
+import java.time.Clock;
+import javax.inject.Inject;
+import play.mvc.Controller;
+import play.mvc.Http;
+import play.mvc.Result;
+import services.settings.SettingsManifest;
+
+public class SessionController extends Controller {
+  private final ProfileUtils profileUtils;
+  private final SettingsManifest settingsManifest;
+  private final Clock clock;
+
+  @Inject
+  public SessionController(
+      ProfileUtils profileUtils, SettingsManifest settingsManifest, Clock clock) {
+    this.profileUtils = profileUtils;
+    this.settingsManifest = settingsManifest;
+    this.clock = clock;
+  }
+
+  public Result extendSession(Http.Request request) {
+    if (!settingsManifest.getSessionTimeoutEnabled()) {
+      return badRequest("Session timeout is not enabled");
```

Comment:
```suggestion
      return badRequest();
```
(blocking) Please remove the message. We don't need to expose the reason to the client.